dialect/sql/builder.go

```diff
@@ -1796,6 +1797,20 @@ func (s *Selector) Having(p *Predicate) *Selector {
 	return s
 }
 
+// ForUpdate appends the `FOR UPDATE` clause to the `SELECT` statement.
+func (s *Selector) ForUpdate() *Selector {
+	s.distinct = false // combination of DISTINCT and FOR UPDATE is not allowed
```

Comment:
this is annoying, but you can't combine DISTINCT and FOR UPDATE. 

I'm not sure how to best fix this as we do distinct by default, I don't really like putting this here in `Selector` ...

I think I'd prefer handling this in `func (q *query) selector() (*sql.Selector, error)`? maybe even error if `Unique` is set ... or would it be okay that `LockForUpdate()` cancels out the Unique?

Reply:
@a8m curious what you think about this?

Reply:
Hey @rubensayshi, I agree with you here. I think the `*sql.Selector` should be simple. 

Regarding this API - I think it's clean, but we should make it possible to support all options (e.g. `FOR SHARE OF <T> NO WAIT`). 

Throwing here a few ideas:

```go
func (s *Selector) For(lockOpts ...sql.LockOptions) *Selector

func (s *Selector) For(strength sql.LockStrength, lockOpts ...sql.LockOptions) *Selector

func (s *Selector) ForShare(lockOpts ...sql.LockOptions) *Selector

func (s *Selector) ForShare(lockCfg sql.LockConfig) *Selector
```

```go
s.For(sql.Share(), sql.Of(t), sql.NoWait())

s.ForShare(sql.Of(t), sql.NoWait())

s.ForShare(sql.LockConfig{
	Tables: []string{t1, t2},
	NoWait: true,
})
```